doc: Update configuration notes for working with IMDSv2 metadata server (#2243)
kubernetes-sigs/aws-load-balancer-controller: docs/deploy/installation.md

    - Ensure subnets are tagged appropriately for auto-discovery to work
    - For IP targets, pods must have IPs from the VPC subnets. You can configure `amazon-vpc-cni-k8s` plugin for this purpose.

## Using metadata server version 2 (IMDSv2)

> If you are using the IMDSv2 you must set the hop limit to 2 or higher in order to allow the AWS Load Balancer Controller to perform the metadata introspection. 
> Otherwise you have to set manually the region, vpc and other in startup parameters.

Reviewer: ```suggestion
If you are using the IMDSv2 you must set the hop limit to 2 or higher in order to allow the AWS Load Balancer Controller to perform the metadata introspection. Otherwise you have to manually specify the AWS region and the VPC via the controller flags `--aws-region` and `--aws-vpc-id`.

```

> How to set the hop limit:
> 
>     ```
>     aws ec2   modify-instance-metadata-options --http-put-response-hop-limit 2 --region <region> --instance-id <instance-id>
>     ```
> 

Reviewer: ```suggestion
!!!tip 
    You can set the IMDSv2 hop limit as follows:
    ```
    aws ec2 modify-instance-metadata-options --http-put-response-hop-limit 2 --region <region> --instance-id <instance-id>
    ```

```
